### PR TITLE
ci(lint): format yaml files

### PR DESCRIPTION
**Pull request by mmorel-35**
Yamllint can help to keep yaml files linted, see [documentation](https://yamllint.readthedocs.io/en/stable/index.html) 

I have just configured it for `.yamllint.yml` and `deployments/charts/kuma/values.yaml`  files to show you the potential.
This is helping to have a README with alphabetically ordered properties.

### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [ ] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: mk/dependencies/yamllint.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+
+OUTPUT_DIR=$1/bin
+VERSION="1.31.0"
+YAMLLINT="${OUTPUT_DIR}"/yamllint.tar.gz
+VERSION_NAME="v${VERSION}"
+
+if [ -z $(pip --version | grep "command not found") ]; then
```
Comment (mmorel-35) [inline mk/dependencies/yamllint.sh:10]: Python and pip are needed to use yamllint. Is it something that should be managed by circleCi directly?
Maybe you also don't want to add Python language. I tried to find if this was possible to use yamllint without Python but couldn't.

File: mk/dependencies/yamllint.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+if [ -x "$(command -v yamllint)" ]; then
+  echo "Yamllint is already installed"
+  exit
+fi
+  echo "Installing Yamllint"
+  set -x
+  sudo apt update --fix-missing
+  sudo env DEBIAN_FRONTEND=noninteractive apt install -y yamllint
+  echo "Yamllint has been installed"
```
Comment (bartsmykla) [inline mk/dependencies/yamllint.sh:13]: This won't work on macOS and other linux distributions, please, could you implement this script to be more OS agnostic?

I would also like to have pinned version of it.

File: mk/dev.mk [hunk 3]
```diff
@@ -63,6 +63,7 @@ GOLANGCI_LINT=$(CI_TOOLS_BIN_DIR)/golangci-lint
 HELM_DOCS=$(CI_TOOLS_BIN_DIR)/helm-docs
 KUBE_LINTER=$(CI_TOOLS_BIN_DIR)/kube-linter
 HADOLINT=$(CI_TOOLS_BIN_DIR)/hadolint
+YAMLLINT=yamllint
```
Comment (bartsmykla) [inline mk/dev.mk:66]: We don't want to use global binaries, we have to figure out the way to put it inside `$(CI_TOOLS_BIN_DIR)`

**Review by bartsmykla — CHANGES_REQUESTED** [review]

**Comment by bartsmykla** [comment]
Hi @mmorel-35. Thank you for the contribution! I like adding the linter, but we have to figure a better way of installing it and putting in appropriate directory.

File: .circleci/config.yml [hunk 4]
```diff
@@ -37,8 +37,8 @@ commands:
                 apt install -y sudo
               fi
 
-              sudo apt update
-              sudo env DEBIAN_FRONTEND=noninteractive apt install -y curl git make unzip gcc xz-utils
+              sudo apt update --fix-missing
+              sudo env DEBIAN_FRONTEND=noninteractive apt install -y curl git make unzip gcc xz-utils python3-pip
```
Comment (mmorel-35) [inline .circleci/config.yml:41]: Hi @bartsmykla ,

The simplest way I could find to stay OS agnostic is using pip/pip3 from python.
I keep getting pip3 command not found or pip command not found.

Any idea what I'm missing here ?

**Comment by mmorel-35** [comment]
Hi @bartsmykla ,
On circleCi, I see the following error but on codespace I don't have this problem.
```
/home/circleci/.kuma-dev/kuma-ci/bin/yamllint .
Traceback (most recent call last):
  File "/home/circleci/.kuma-dev/kuma-ci/bin/yamllint", line 5, in <module>
    from yamllint.cli import run
ModuleNotFoundError: No module named 'yamllint'
```
Any Idea what would be the difference between the two environnements ?

**Comment by lahabana** [comment]
Not a fan of not being able to run this linter as part of `make check` this means you'd open the PR, then check the yaml is not right so fix it and then push again.

**Review by lahabana — CHANGES_REQUESTED** [review]
Ok so I looked at what this does. The only lint it finds is with values.yaml file. What it does is reordering the keys. I have to admit that I'm not a fan of this because atm values.yaml is ordered in a logical.

With a quick test I've noticed that if we want formatted yamls `yq` (that we already have) should be enough.


Here's a quick test:
```
➜  kuma git:(fix6985) ✗ find . -name '*.yaml' -o -name '*.yml' -type f | grep -v deployments | grep -v app/kumactl/data/install | grep -v tools/policy-gen/templates | grep -v 'golden.yaml' | grep -v 'template.yaml' | xargs -n 1 ~/.kuma-dev/kuma-master/bin/yq eval -i
```

**Comment by slonka** [comment]
@mmorel-35 - re-posting `lahabana` Q:

> WDYT about doing something like this instead?

**Comment by mmorel-35** [comment]
Hi @slonka , @lahabana ,

Yq could do the job for ordering the files, but I'm not sure you can give it a config like you are able to do it with yamllint.

If you think it's enough I can give it a try. How shall the make command be named ?

**Comment by lahabana** [comment]
I'd go for `fmt/yaml` and put it as dep of `format/common`

**Comment by slonka** [comment]
This got a bit stale, and it will keep getting stale until we merge it because the changes in these files are quite frequent. @lahabana can you voice your opinion on the overall approach of this PR and then we can get this merged or changed, otherwise this will be stuck forever.

**Pull request closed without merging**